site/src/AppRouter.tsx

```diff
@@ -309,3 +338,14 @@ export const AppRouter: FC = () => {
     </Suspense>
   )
 }
+
+// AuthenticatedProviders are used to provide authenticated contexts to children
+export const AuthenticatedProviders: FC = () => {
+  return (
+    <DashboardProvider>
+      <ProxyProvider>
+        <Outlet />
+      </ProxyProvider>
+    </DashboardProvider>
+  )
+}
```

Comment:
IMO, having a `RequireAuth` and `AuthenticatedProviders` is kinda confusing because RequireAuth should do that. I think the best would be to move the ProxyProvider inside the DashboardProvider and wrap the component there or even merge the DashboardProvider with ProxyProvider if that is only used inside the Dashboard.

Reply:
It is also used in the Terminal Page, and the settings page